script/fetch-issues.js

```diff
@@ -141,23 +192,132 @@ async function fetchNextPage(owner, name, isIssue, data, cursor, pageCount) {
+              }
+              return comment;
+            });
+          }
+        } else {
+          // for debug
+          // console.log('pull request does not have reactions. repoOwner:', owner,
+          //   'repoName:', name, 'pullRequest number:', node.number);
+        }
+
+        // filter out useless reviews
+        rawComments = rawComments.filter(node => {
+          let flag = true;
+          // filter comments that don't need meta-reviews
+          for (const ignoreContent of ignoreReviewContent.split(' ')) {
+            if (node.bodyText.match(ignoreContent)) {
+              // filter reviews with specific content
+              flag = false;
+            }
+          }
+          for (const ignoreAuthor of ignoreReviewAuthor.split(' ')) {
+            if (node.author && node.author.login === ignoreAuthor) {
+              // filter reviews done by specific authors
+              flag = false;
+            }
+          }
+          return flag;
+        });
+
+        comments = rawComments.map(node => {
+          let commentAuthor, reactions;
+
+          if (node.author) {
+            commentAuthor = {
+              login: node.author.login,
+              avatarUrl: node.author.avatarUrl,
+              name: node.author.name
+            };
+          } else {
+            commentAuthor = {
+              login: null,
+              avatarUrl: null
+            };
+          }
+
+          if (node.reactions) reactions = node.reactions.nodes;
+
+          return {
+            id: node.id,
+            url: node.url,
+            bodyText: node.bodyText,
+            diffHunk: node.diffHunk ? node.diffHunk : null,
+            author: commentAuthor,
+            reactions,
+            createdAt: node.createdAt,
+            lastEditedAt: node.lastEditedAt,
+            updatedAt: node.lastEditedAt ? node.lastEditedAt : node.createdAt
+          };
+        });
+        info.issue.pullRequest = {
+          htmlUrl: node.url,
+          comments
         };
       }
       return info;
-    });
+    }));
     data = data.concat(result);
-    if (hasPreviousPage && (pageCount < pageThreshold || pageThreshold == -1)) {
+    if (hasPreviousPage && !reachDateThreshold &&
+       (pageCount < pageThreshold || pageThreshold == -1)) {
       return fetchNextPage(owner, name, isIssue, data,
-        cursor, pageCount);
+        cursor, pageCount, 0);
     } else {
       return data;
     }
   } else {
     console.log('Warning: no available data. owner:', owner, 'name:', name,
-      'isIssue:', isIssue, 'pageCount:', pageCount);
-    return data;
+      'isIssue:', isIssue, 'pageCount:', pageCount, 'error message:', errors);
+    warningCnt += 1;
+    sleep.sleep(3);
+    if (warningCnt < 15) {
+      console.log('warning count:', warningCnt);
+      return fetchNextPage(owner, name, isIssue, data, cursor, pageCount, warningCnt);
+    } else {
+      console.log('exit');
```

Comment:
Not very clear what this means.

Reply:
Yeah, I just amended my commit. Meant to say, when the number of "no data" warnings exceeds the threshold, stop fetching on this repo.